authentication/pipeline/user.py

```diff
@@ -108,6 +109,44 @@ def require_password_and_profile(
     })
 
 
+@partial
+def require_profile_update_user_via_saml(
+        strategy, backend, user=None, is_new=False, *args, **kwargs
+):  # pylint: disable=unused-argument
+    """
+    Sets a new user's password and profile, and updates the user first and last names.
+    Touchstone only returns DisplayName, which is the full name, and that is what the
+    user first_name is initially set to.
```

Comment:
What happens if the touchstone user's full name is greater than 30 characters (the limit on the field)? We should also check this below and add long names to our tests

Reply:
I modified the name assignment code to handle this and truncate first and last names if they are more than 30 characters; tests included.

Reply:
I'm not sure 30 characters is a reasonable limit for a full name...

Reply:
That's 30 characters per first name, 30 per last name: these are the defaults for the `django.contrib.auth.models.User` model (though last_name goes up to 150 in Django 2.0)